cli.py

```diff
@@ -15,18 +18,95 @@ def write_dict(file_path, data_dict):
+    
+    if args['biosamples']:
+        if not data:
+            print('No Data to Submit to BioSamples')
+            sys.exit(2)
 
-    if data:
-        json_file_path = os.path.splitext(excel_file_path)[0] + '.json'
+        if issues:
+            user_text = input(f'{len(issues)} issues detected. Continue with BioSamples Submission? (y/N)?:')
+            if not user_text.lower().startswith('y'):
+                print('Exiting')
+                sys.exit(0)
+
+        aap_url = args['aap_url']
+        url = args['biosamples_url']
+        domain = args['biosamples_domain']
+
+        if 'AAP_USERNAME' in os.environ:
+            aap_username = os.environ['AAP_USERNAME']
+        else:
+            print('No AAP_USERNAME detected in os environment variables.')
+            sys.exit(2)
+
+        if 'AAP_PASSWORD' in os.environ:
+            aap_password = os.environ['AAP_PASSWORD']
+        else:
+            print('No AAP_PASSWORD detected in os environment variables.')
+            sys.exit(2)
+
+        print(f'Attempting to Submit to BioSamples: {url}, AAP: {aap_url}')
+        aap_client = AapClient(url=aap_url, username=aap_username, password=aap_password)
+        biosamples = BioSamples(aap_client, url, domain)
+        sample_count = 0
+        error_count = 0
+        for row in data:
+            if 'sample' in row:
+                try:
+                    # ToDo: Maybe store this in row['sample']['request']
+                    row['sample_request'] = biosamples.encode_sample(row['sample'])
+                    sample_count = sample_count + 1
+                except Exception as error:
+                    # ToDo: Also write to issues
+                    # ToDo: Maybe append this to row['sample']['errors']
+                    # row['sample'].setdefault('errors', []).append()
+                    row.setdefault('sample_request', {})['error'] = f'Encoding Error: {error}'
+                    error_count = error_count + 1
         write_dict(json_file_path, data)
+        print(f'Data from {sample_count} BioSamples conversions and {error_count} errors written to: {json_file_path}')
+
+        biosample_count = 0
+        error_count = 0
+        for row in data:
+            if 'sample_request' in row and 'error' not in row['sample_request']:
+                try:
+                    # ToDo: Maybe store this in row['sample']['response']
+                    row['sample_response'] = biosamples.send_sample(row['sample_request'])
+                    biosample_count = biosample_count + 1
+                except Exception as error:
+                    # ToDo: Also write to issues
+                    # ToDo: Maybe append this to row['sample']['errors']
+                    # row['sample'].setdefault('errors', []).append()
+                    row.setdefault('sample_response', {})['error'] = f'BioSamples Error: {error}'
+                    error_count = error_count + 1
+        write_dict(json_file_path, data)
```

Comment:
This would overwrite the file that has been created in line 91

Reply:
Yep, that is intended, to update the file with the latest information.